jumanji/environments/routing/rware/utils.py

```diff
@@ -0,0 +1,865 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+from typing import Tuple, Union
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from jumanji.environments.routing.rware.types import (
+    _AGENTS,
+    _SHELVES,
+    Action,
+    Agent,
+    Direction,
+    Entity,
+    Position,
+    Shelf,
+)
+from jumanji.tree_utils import tree_add_element, tree_slice
+
+
+def get_entity_ids(entities: Entity) -> chex.Array:
+    """Get ids for agents/shelves.
+
+    Args:
+        entities: a pytree of Agent or Shelf type.
+
+    Returns:
+        an array of ids.
+    """
+    return jnp.arange(entities[1].shape[0])
```

Comment:
Not sure if there's a cleaner way to get the length of a stacked pytree :thinking:

Maybe `jnp.arange(len(entities[1]))` but it's honestly not much better